pandas/core/indexes/datetimes.py

```diff
@@ -262,6 +264,14 @@ class DatetimeIndex(DatelikeOps, TimelikeOps, DatetimeIndexOpsMixin,
     month_name
     day_name
 
+    Examples
+    --------
+    >>> dates = pd.DatetimeIndex(freq='M', start='2018-3-10', periods=6)
+    >>> dates
+    DatetimeIndex(['2018-03-31', '2018-04-30', '2018-05-31',
+                   '2018-06-30', '2018-07-31', '2018-08-31'],
+                  dtype='datetime64[ns]', freq='M')
```

Comment:
`Examples` should be the last section.

As a convention, we name indices with `idx`, I think this would be preferred over `dates`. In this case, as `dates` is not being used more than to display it, I wouldn't store the index in a variable but show it directly

The example looks great, but I think couple of more examples showing the contructor with `end`, `closed`, and may be a different frequency, could be useful.